x/oracle/abci.go

```diff
@@ -56,6 +63,16 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) error {
 			if err = k.SetExchangeRateWithEvent(ctx, ballotDenom.Denom, exchangeRate); err != nil {
 				return err
 			}
+
+			// Stamp rate every stamp period
+			if isPeriodLastBlock(ctx, params.StampPeriod) {
+				k.AddHistoricPrice(ctx, ballotDenom.Denom, exchangeRate)
+			}
+
+			// Set median price for each denom every median period
+			if isPeriodLastBlock(ctx, params.MedianPeriod) {
+				k.SetMedian(ctx, ballotDenom.Denom)
```

Comment:
how expensive this is? Should not be merged if we don't have a benchmark. Can be commented out and done later.

Reply:
Commented out and made issue for benchmarking median calculating and setting 👍